src/overlays/actors/ovl_Bg_Gnd_Darkmeiro/z_bg_gnd_darkmeiro.c

```diff
@@ -35,25 +34,189 @@ const ActorInit Bg_Gnd_Darkmeiro_InitVars = {
+        this->actionState |= 2;
+    }
+}
+
+void BgGndDarkmeiro_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+    s32 local_c = 0;
+    s32 pad;
+
+    this->actionFunc = BgGndDarkmeiro_DoNothing;
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+
+    switch (thisx->params & 0xFF) {
+        case MODE_ZERO:
+            this->dyna.actor.draw = BgGndDarkmeiro_Draw0;
+            this->dyna.actor.flags |= 0x80;
+            break;
+        case CLEAR_BLOCK:
+            DynaPolyInfo_Alloc(&D_0600C080, &local_c);
+            this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, local_c);
+
+            if (((this->dyna.actor.params >> 8) & 0x3F) == 0x3F) {
+                this->actionFunc = BgGndDarkmeiro_StaticBlock;
+                this->dyna.actor.draw = BgGndDarkmeiro_DrawStaticBlock;
+            } else {
+                this->actionState = this->timer1 = this->timer2 = 0;
+                thisx->draw = BgGndDarkmeiro_DrawSwitchBlock;
+                this->actionFunc = BgGndDarkmeiro_SwitchBlock;
+                if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F)) == 0) {
+                    func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+                } else {
+                    this->timer1 = 64;
+                    this->actionState |= 2;
+                }
+            }
+            break;
+        case BLOCK_TIMER:
+            this->actionState = this->timer1 = this->timer2 = 0;
+            this->actionFunc = BgGndDarkmeiro_BlockTimer;
+            thisx->draw = NULL;
+            if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 1)) {
+                this->timer1 = 64;
+                this->actionState |= 4;
+            }
+            if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 2)) {
+                this->timer2 = 64;
+                this->actionState |= 8;
+            }
+            if ((this->timer1 != 0) || (this->timer2 != 0)) {
+                Flags_SetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F));
+            } else {
+                Flags_UnsetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F));
+            }
+            break;
+    }
+    if (globalCtx) {}; // needed for matching
+}
+
+void BgGndDarkmeiro_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgGndDarkmeiro* this = THIS;
+
+    if ((this->dyna.actor.params & 0xFF) == 1) {
+        if (globalCtx) {}; // needed for matching
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+void BgGndDarkmeiro_DoNothing(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+}
+
+void BgGndDarkmeiro_BlockTimer(BgGndDarkmeiro* this, GlobalContext* globalCtx) {
+    s16 timeLeft;
+
+    if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 1)) {
```

Comment:
`(this->dyna.actor.params >> 8) & 0x3F` is the same as `this->dyna.actor.params >> 8 & 0x3F` (you get used to it and it's a lot easier to read)

Reply:
I don't have C operator preference memorized, so I'd prefer to keep the (). Also this should just be SWITCH_FLAG or something anyways.

Reply:
yeah i prefer parens myself. its a topic a bunch of us disagree on lmao
but yeah petrie is right, these will be defines anyway so in this specific case i wouldnt worry